projects/core/directives/hint/hint-hover.directive.ts

```diff
@@ -29,6 +29,7 @@ export class TuiHintHoverDirective extends TuiDriver {
             repeat(),
         ),
     ).pipe(
+        map(value => !tuiGetElementObscures(this.el.nativeElement)?.length && value),
```

Comment:
Change the order so the easiest to compute condition is first. This way we will not need to compute obscurers for falsy values. Also lets move it below `filter`.